### ISSUE TITLE
Subscription connection handler

### ISSUE THREAD

**Issue by alexeypolitov** [awslabs/aws-mobile-appsync-sdk-ios#68]
Do you know some way to handle subscription success connection?
I was made some changes in `AWSAppSyncSubscriptionWatcher ` to handle this, but maybe you know better way

```
    private func startSubscription()  {
        let semaphore = DispatchSemaphore(value: 0)
        
        self.performSubscriptionRequest(completionHandler: { [weak self] (success, error) in
            if let error = error {
                self?.resultHandler(nil, nil, error)
            }
            semaphore.signal()
+            if let handle = self?.completeHandler {
+                self?.handlerQueue.async {
+                    handle()
+                }
+            }
        })
        
        semaphore.wait()
    }
```

**Comment by rohandubal** [comment]
Hello @alexeypolitov 

Is your use-case to get a callback when the subscription is active on the client?

Thanks,
Rohan

**Comment by alexeypolitov** [comment]
@rohandubal resolved in pull #75
